samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -429,6 +429,25 @@ private void writeContainerPlacementResponseMessage(ContainerPlacementRequestMes
             responseMessage, System.currentTimeMillis()));
   }
 
+  /**
+   * Gets the hostname on which container is either currently running or was last seen on if it is not running
+   */
+  private String getSourceHostForContainer(ContainerPlacementRequestMessage requestMessage) {
+    String sourceHost = null;
+    String processorId = requestMessage.getProcessorId();
+    if (samzaApplicationState.runningProcessors.containsKey(processorId)) {
+      SamzaResource currentResource = samzaApplicationState.runningProcessors.get(processorId);
+      LOG.info("Processor ID: {} matched a running container with containerId ID: {} is running on host: {} for ContainerPlacement action: {}",
+          processorId, currentResource.getContainerId(), currentResource.getHost(), requestMessage);
+      sourceHost = currentResource.getHost();
+    } else {
+      sourceHost = samzaApplicationState.jobModelManager.jobModel().getContainerToHostValue(processorId, SetContainerHostMapping.HOST_KEY);
+      LOG.info("Processor ID: {} is not running and was last seen on host: {} for ContainerPlacement action: {}",
+          processorId, sourceHost, requestMessage);
```

Comment:
CPM does a similar thing at 
```
String lastSeenOn = state.jobModelManager.jobModel().getContainerToHostValue(processorId, SetContainerHostMapping.HOST_KEY);
```
Is it possible to consolidate these multiple job-model-manager accesses?

maybe call it just getHostForContainer?

Reply:
maybe link jira for that here?

Reply:
sure